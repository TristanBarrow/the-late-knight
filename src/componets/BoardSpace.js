import React from 'react';

class BoardSpace extends React.Component {
  render() {
    return (
      <button className='board-space'>
        {this.props.data}
      </button>
    )
  }
}

export default BoardSpace;