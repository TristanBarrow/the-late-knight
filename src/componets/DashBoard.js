import React from 'react'
import ChessBoard from './Board';

const DashboardPage = () => (
  <div>
    <ChessBoard />
  </div>
);

export default DashboardPage;