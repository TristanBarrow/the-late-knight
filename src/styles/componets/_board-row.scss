.board-row {
  width: $board-size;
  height: $space-size;
}