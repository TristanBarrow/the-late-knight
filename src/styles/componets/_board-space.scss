.board-space {
  width: $space-size;
  height: $space-size;
  background: none;
  border: 1px solid $gray;
  &:active {
    background: $off-white;
    border: 1px solid $gray;
  }
}