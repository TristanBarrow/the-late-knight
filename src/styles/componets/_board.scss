.board {
  border: 1px solid $gray;
  padding: 0;
  margin: 0;
  height: $board-size;
  width: $board-size;
}