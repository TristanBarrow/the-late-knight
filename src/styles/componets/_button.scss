.button {
  border: 0;
  background: $blue;
  padding: $s-size;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  color: white;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
  color: white;
}

.button--gray {
  background: $light-gray;
}