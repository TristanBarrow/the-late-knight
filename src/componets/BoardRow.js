import React from 'react';
import BoardSpace from './BoardSpace';

const BoardRow = ({ row }) => (
  <div className='board-row'>
    {
      row.map((space) => {
        return <BoardSpace key={space} data={space}/>;
      })
    }
  </div>
);

export default BoardRow;