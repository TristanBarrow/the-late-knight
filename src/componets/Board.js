import React from 'react';
import BoardRow from './BoardRow';
import boardLocations from '../setup/boardData';

class ChessBoard extends React.Component {
  render() {
    let id = 0;
    return (
      <div className='content-container'>
        <div className='board'>
          {
            boardLocations.map((row) => {
              return <BoardRow key={id++} row={row}/>;
            })
          }
        </div>
      </div>
    )
  }
}

export default ChessBoard;