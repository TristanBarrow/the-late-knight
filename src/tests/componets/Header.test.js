import React from 'react';
import { shallow } from 'enzyme'; 
import { Header } from '../../componets/Header';

test('Render the Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('Call start logout on click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});

