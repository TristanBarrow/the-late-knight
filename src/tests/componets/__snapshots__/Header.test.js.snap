// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Template
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
